templates/Procfile

```diff
@@ -1,2 +1,2 @@
-web: bundle exec unicorn -p $PORT -c ./config/unicorn.rb
+web: bundle exec puma -C ./config/puma.rb
```

Comment:
> . I would have guessed Heroku would have assigned it as a default somewhere

heroku sets `PORT` environment variable. You're missing the port directive in your config/puma.rb file


```ruby
port ENV['PORT']     || 3000
```

Without it puma doesn't know what port to use.